lib/spack/spack/package.py

```diff
@@ -888,7 +888,7 @@ def extends(self, spec):
         if spec.name not in self.extendees:
             return False
         s = self.extendee_spec
-        return s and s.satisfies(spec)
+        return s and spec.satisfies(s)
```

Comment:
This is called with `strict=False` implicitly so I'm surprised that the order matters. Is there an issue referring to the problem that comes up for `s.satisfies(spec)`?

Reply:
My initial thought here is that if we have some spec `X` and want to know if we extend it, our extendee `E` should probably satisfy it. I'd be curious to know the case where it doesn't.

Reply:
I found this when trying to write a test for the fsview extensions. When querying with `spack extensions python@2.7.13`, I got nothing. Switching this around got me the list I expected.

Reply:
Also, this makes more sense in the sense that it now says "if the spec you asked to list extensions for satisfies the spec given in this package, this package extends the spec you gave". The other way reads backwards. Or is the method itself backwards?